### DISCUSSION TITLE
Domain decomposition

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#61] [General discussion]
Hello,

I am trying to use the new functionality for domain decomposition for the _pipflow_ case but I do not see my desired decomposition. This is what I am setting in the config.xml file:

```xml
<domain>
    ...
    <mABx>15</mABx>
    <mABy>8</mABy>
    <mABz>2</mABz>
</domain>
```

What I get is (10x6x4):
![Screenshot 2022-08-04 174109](https://user-images.githubusercontent.com/15649053/182959697-040ca9f9-fc73-496b-83cf-01be6ad3d665.png)

**Comment by mojtabaAmiraslanpour** [comment]
Any thoughts on this?

**Comment by gzavo** [comment]
Hi @mojtabaAmiraslanpour,

I'm not sure why you get this decomposition. This is the relevant code:
https://github.com/UvaCsl/HemoCell/blob/ccccc63aacc211847e79f4478cf701e78663487b/core/hemoCell.cpp#L447-L460

It can be that this does not play nice together with the voxelizer. Maybe try another example where no voxelization is used, just to see if this is the case.

**Comment by mojtabaAmiraslanpour** [comment]
I managed to make it work. I was assuming it is already added to the available examples. I used initializeLattice and everything worked just fine.
